mac/KeymanEngine4Mac/KeymanEngine4Mac/KME/OnScreenKeyboard/KeyView.m

```diff
@@ -232,17 +232,14 @@ - (void)mouseUp:(NSEvent *)theEvent {
 }
 
 -(void)processKeyClick {
-    ((void (*)(id, SEL))[self.target methodForSelector:self.action])(self.target, self.action);
+    SEL selector = self.action;
+    IMP imp = [self.target methodForSelector:selector];
+    void (*func)(id, SEL, id) = (void *)imp;
```

Comment:
I guess hard casting a function prototype caused this problem in the first place. Is there any way we can avoid a hard cast, or document it more clearly?

Reply:
I talked to ChrisH about it and he suggested that there were probably several better approaches. I agree. As it is I only barely understand this code, and given where were are in the game, I was reluctant to try anything more aggressive.